### PR TITLE
adds host to formals of install_theme() and new_site()

### LINKED ISSUE #264: Consider a host argument for install_theme()?
I'd be happy to make a PR for this.

I have GitHub Enterprise (GHE) at work, and I have an ambition to produce a hugo theme for my company. I would like to be able to host this on our instance of GHE.

Further, I would like to offer a company-specific function `acmetools::install_theme_acme()` that would wrap `blogdown::install_theme()`, but it would require a host argument that might look something like:

```R
install_theme <- function(theme, ..., host = "github.com"){
  ...
}
```

This way, I could design `acmetools::install_theme_acme()` to look like:

```R
install_theme_acme <- function(theme, ..., host = "github.acme.com"){
  blogdown::install_theme(theme, ..., host = host)
}
```

What do you think?

If acceptable, perhaps `new_site()` could be adapted, as well? Others?

### PR DESCRIPTION
**Pull request by ijlyttle**
This is just to open the conversation - not finished yet. When done, this fixes #264.

When I add the argument `host`, I get a warning on R CMD check:

```
* checking Rd \usage sections ... WARNING
Duplicated \argument entries in documentation object 'hugo_cmd':
  ‘host’
``` 

To get around this, I could use `hostname` instead.

### CODE REVIEW

**Comment by ijlyttle** [comment]
The coding is complete (I hope) - it will take me a little time to make sure that everything *actually* works.

File: R/hugo.R [hunk 1]
```diff
@@ -92,7 +94,8 @@ change_config = function(name, value) {
 #' if (interactive()) new_site()
 new_site = function(
   dir = '.', install_hugo = TRUE, format = 'toml', sample = TRUE,
-  theme = 'yihui/hugo-lithium-theme', theme_example = TRUE, serve = interactive()
+  theme = 'yihui/hugo-lithium-theme', theme_example = TRUE, serve = interactive(),
+  hostname = 'github.com'
```
Comment (yihui) [inline R/hugo.R]: I think it makes a little more sense to move this argument closer to `theme` (e.g. right after `theme`).

Comment (ijlyttle) [reply R/hugo.R]: will do!

**Comment by ijlyttle** [comment]
I will adjust the roxygen process - if it does not do what I want it to do, I will leave out the changes in Rd files.

**Comment by ijlyttle** [comment]
@yihui - I have rebuilt the .Rd files - just to be sure that the result is as you expect - thanks!

**Comment by ijlyttle** [comment]
I gave it a quick test, and it works as expected for me, both for GitHub and GHE. Will make those additions right away. Thanks!

**Comment by ijlyttle** [comment]
should be good-to-go!

File: NEWS.md [hunk 2]
```diff
@@ -2,7 +2,9 @@
 
 ## NEW FEATURES
 
-- Added a `title_case` argument to the `new_post()` function; if \code{TRUE}, the post title will be converted to title case. See `?blogdown::new_post` for details.
+- Added a `title_case` argument to the `new_post()` function; if `TRUE`, the post title will be converted to title case. See `?blogdown::new_post` for details.
```
Comment (yihui) [inline NEWS.md]: Nice catch. Thanks!

**Pull request merged**